src/types/constraint.ts

```diff
@@ -1,12 +1,19 @@
 import { Runtype, Rt, Static, create, validationError } from '../runtype'
 import { String } from './string'
 
+export type ConstraintCheck<T> = (x: T) => boolean | string
+export type ConstraintCorrection<T> = (x: T) => T
+export type ConstraintRightInverse<T> = (x: T) => T
+
 export interface Constraint<A extends Rt> extends Runtype<Static<A>> {
   tag: 'constraint'
-  underlying: A
+  underlying: A,
+  constraint: ConstraintCheck<A>,
+  correction?: ConstraintCorrection<A>,
+  rightInverse?: ConstraintRightInverse<A>
 }
 
-export function Constraint<A extends Rt>(underlying: A, constraint: (x: A) => boolean | string) {
+export function Constraint<A extends Rt>(underlying: A, constraint: ConstraintCheck<A>, correction?: ConstraintCorrection<A>, rightInverse?: ConstraintRightInverse<A>) {
```

Comment:
This looks like a bug carried over from the previous code, but shouldn't this be

```ts
export function Constraint<A extends Rt>(underlying: A, constraint: ConstraintCheck<Static<A>>, correction?: ConstraintCorrection<Static<A>>, rightInverse?: ConstraintRightInverse<Static<A>>) {
```